### PR TITLE
Social Slice

### PR DESCRIPTION
**Pull request by JoeStanton**
### Motivation

According to designs in Zeplin: https://app.zeplin.io/project.html#pid=57c8297fc3cf63795b6a4520&dashboard

### Test plan

- TODO: Write a test plan.

### Pre-merge checklist

- [ ] Documentation
- [ ] Unit tests
- [ ] Reviews
  - [ ] Code review
  - [ ] Design review
- [ ] Manual testing
  - [ ] Windows 7 IE 11
  - [ ] Windows 10 Edge
  - [ ] Mac OS El Capitan Safari
  - [ ] Mac OS El Capitan Chrome
  - [ ] Mac OS El Capitan Firefox
  - [ ] iOS 9 Safari
  - [ ] Android 6 Chrome
  - [ ] Listen to the site on a screen reader
  - [ ] Navigate the site using the keyboard only
- [ ] Tester approved

### CODE REVIEW

File: site/pages/about-us/index.js [hunk 1]
```diff
@@ -2,11 +2,13 @@ import React from 'react';
 
 import TimelineSlice from './timeline-slice';
 import PrinciplesSlice from './principles-slice';
+import SocialSlice from './social-slice';
 import ContactUs from '../../slices/contact-us-slice';
 
 const AboutUs = ({ contactUsURL }) => {
   return (
     <div>
+      <SocialSlice />
```
Comment (JoeStanton) [inline site/pages/about-us/index.js]: This'll move to the bottom of the page, just here for convenience during dev

File: site/pages/about-us/social-slice/index.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    <div className={styles.handle}>
+      <InlineSVG src={twitterIconSVG} className={styles.twitterIcon} />
+      {handle}
+    </div>
+    <div className={styles.tweet}>
+      {tweet}
+    </div>
+    <div className={styles.meta}>
+      <span className={styles.retweets}>
+        <InlineSVG src={retweetIconSVG} className={styles.icon} />
```
Comment (JoeStanton) [inline site/pages/about-us/social-slice/index.js]: Currently no PNG fallback

**Comment by samsmith1983** [comment]
This doesn't look ideal on iPad (iPad Air with iOS9):

social_ipad

**Comment by samsmith1983** [comment]
Layout issue detailed above is also occurring on IE11 (Windows 7).

File: site/css/_colors.css [hunk 3]
```diff
@@ -19,6 +19,7 @@
 @value cainYellow: #FFD811;
 @value cainYellowFaded: #fede3f;
 @value scorpian: #606060;
+@value yoloPurple: #B88DFF;
```
Comment (asavin) [inline site/css/_colors.css]: Potentially need to update this page with new colours http://brand.red-badger.com/d/WEl30uS4YzKw/red-badger-styleguide-style-guide#/elements/colours

File: site/fetchers/twitter.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    timeout: 10000,
+    body: 'grant_type=client_credentials',
+  })
+    .then(handleErrors)
+    .then(response => response.json())
+    .then(response => response.access_token);
+
+/**
+ * The URL of the Tweet is usually the property tweet.entities.url[0].url
+ * However if it's a re-tweet then the entities.url is an empty array and the URL is contained in
```
Comment (JRGranell) [inline site/fetchers/twitter.js]: Done

File: site/css/_colors.css (hunk 3 above)
Comment (JRGranell) [inline site/css/_colors.css]: Renamed yoyoPurple to keep consistent with the style guid

**Comment by samsmith1983** [comment]
I know JR's already aware of this, but writing it here so i don't forget:

The feed starts with instagram items on Chrome (OSX and Android) Firefox, but starts with twitter items on Safari (OSX and iOS), IE11, Edge.

**Comment by JRGranell** [comment]
@samsmith1983 does that also happen when turning JS off? I wonder if it's the JS that is changing the order, or whether the browser renders the order differently.

**Comment by samsmith1983** [comment]
@JRGranell Yep I think you're on to something. If I turn off javascript in Safari, it's goes the other way round (to match Chrome). 

Do we know which way is correct?

**Comment by JRGranell** [comment]
@samsmith1983 that's interesting and also a bit strange, it should be ordered by date, regardless of the source, with the most recent post/tweet on the left.

**Comment by samsmith1983** [comment]
OK, will try to figure out the correct order... The tweets are actually in the opposite order in the above browser camps, too.

**Comment by samsmith1983** [comment]
Another issue: With the current content, the _"RT @badgercast: First episode..."_ tweet contains a link to the current page, rather than to the tweet. @JRGranell

**Comment by JRGranell** [comment]
Good spot on the incorrect URL. Re-tweets use a different property on the data that we get, I'll fix it.

**Comment by samsmith1983** [comment]
Ok so with the current content, this is should be the order:

INSTAGRAM Olga Kanban: 28/11/2016 13:19
INSTAGRAM ranj keyboard: 17/12/2016 18:44
INSTAGRAM Stu drinking: 19/12/2016 21:20
INSTAGRAM Milo: 06/01/2017 15:00
TWITTER "Thanks @CreativePool...": 9/01/2017 3:57
TWITTER "Great tutorial on building": 9/01/2017 8:14
INSTAGRAM ASOS: 10/1/2017 17:07
TWITTER "New Year.. New job:: 11/01/2017 8:14
TWITTER "RT @badgercast...": 12/01/2017 3:03

**Comment by samsmith1983** [comment]
Order issue is now fixed 🎉

**Comment by samsmith1983** [comment]
@JRGranell There's still more that needs to be done with accessibility:

- Screenreader-wise, it doesn't seem to be possible to get past the first item (using OSX voiceover and Chrome at least).
- Keyboard navigation-wise, Instagram posts seem to work well - links get highlighted. But with Twitter posts, its difficult to tell what is selected as the highlighting seems to be hidden.

**Comment by samsmith1983** [comment]
@JRGranell If we go with the "non-javascript" option, we'll need a way to scroll through the items on desktop.

**Comment by samsmith1983** [comment]
Also, with javascript off on mobile is there a way we can get the scrolling to snap to each item?

**Comment by samsmith1983** [comment]
A side-effect of the scroll bars (when javascript is turned off) is that there's a gap between the bottom of the social slice and the page footer. This gap is visible with javascript turned on. Need to check that @sarigriffiths is happy with this before deploying:

screen shot 2017-01-17 at 14 40 38

**Comment by samsmith1983** [comment]
One thing missing from the design is underlined links in the tweets. 

Should look like this:
screen shot 2017-01-18 at 10 22 15

**Comment by samsmith1983** [comment]
^ it's been agreed previously that we won't have underlined links, as the whole link takes the user to twitter anyway.

**Pull request merged**